tests/TestCase/ORM/QueryTest.php

```diff
@@ -3466,4 +3466,28 @@ public function testAllNoDuplicateTypeCasting()
 
         $this->assertEquals(1, $query->__debugInfo()['decorators'], 'Only one typecaster should exist');
     }
+
+    /**
+     * Tests that selectAllExcept() selects all columns, except
+     * the one that are passed.
+     *
+     * @return void
+     */
+    public function testSelectAllExcept()
+    {
+        $table = TableRegistry::get('articles');
+        $result = $table
+            ->find()
+            ->selectAllExcept(['body']);
```

Comment:
Would be good to have a test covering how this method is expected to work when it is called multiple times.